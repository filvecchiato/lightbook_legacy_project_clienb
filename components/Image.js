import React from 'react';
import { StyleSheet, ImageBackground } from 'react-native';

const UserImage = ({ image }) => {
  return <ImageBackground source={{ uri: image.url }} style={styles.image} />;
};

const styles = StyleSheet.create({
  image: {
    aspectRatio: 1 * 1.7,
    resizeMode: 'cover',
  },
});

export default UserImage;
