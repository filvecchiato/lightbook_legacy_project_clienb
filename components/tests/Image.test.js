import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../Image';

test('renders correctly', () => {
  const tree = renderer.create(<Image image={{ url: 'test' }} />).toJSON();
  expect(tree).toMatchSnapshot();
});
