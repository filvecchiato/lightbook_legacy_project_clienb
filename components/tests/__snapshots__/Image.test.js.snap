// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": 740,
      "marginVertical": 5,
      "resizeMode": "cover",
      "width": 1334,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "test",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 740,
          "width": 1334,
        },
        undefined,
      ]
    }
  />
</View>
`;
